arbitrator/wasm-libraries/user-host-trait/src/lib.rs

```diff
@@ -477,6 +477,27 @@ pub trait UserHost: GasMeteredMachine {
         trace!("account_codehash", self, address, hash)
     }
 
+    fn account_code(&mut self, address: u32, dest: u32, size: u32) -> Result<(), Self::Err> {
+        self.buy_ink(HOSTIO_INK + EVM_API_INK)?;
+        let address = self.read_bytes20(address)?;
+
+        let (code, gas_cost) = self.evm_api().account_code(address);
+        assert!(code.len() <= size as usize);
+        self.write_slice(dest, &code)?;
+
+        self.buy_gas(gas_cost)?;
+        trace!("account_code", self, address, be!(size))
+    }
+
+    fn account_code_size(&mut self, address: u32) -> Result<u32, Self::Err> {
+        self.buy_ink(HOSTIO_INK + EVM_API_INK)?;
```

Comment:
We should probably require a sentry for these or at the very least `params.ExtcodeCopyBaseEIP150`

Reply:
In Ethereum a sentry is an amount of gas you must have, but needn't pay, to do something. You can use `require_ink` and `require_gas` to achieve this. It's useful in contexts where you're about to do work that might be expensive, or if you want to restrict the kinds of things one can do with a call stipend (another EVM concept: a small gas credit you get when transferring value but for which we only want nonmutating purchases)